Sources/Workspace/Workspace.swift

```diff
@@ -1636,9 +1628,24 @@ extension Workspace {
 
                             switch extractResult {
                             case .success:
-                                if let expectedPath = self.path(for: artifact), !self.fileSystem.isDirectory(expectedPath) {
-                                    tempDiagnostics.emit(.artifactNotFound(targetName: artifact.targetName, artifactName: expectedPath.basename))
+                                let extractedFiles = tempDiagnostics.wrap { try self.fileSystem.getDirectoryContents(parentDirectory)
+                                    .map{ parentDirectory.appending(component: $0) }
+                                    .filter{ $0 != archivePath }
```

Comment:
Extracting to the same directory seems reasonable for now, and looks like how it already was.  Might be useful to extra to a different directory in the future, but I don't see any problem with the existing approach other than having to filter (especially if we expect it to only have the one top-level entry).

Reply:
right, I kept the same behavior. long run this should also be cached in the user level cache